packages/teams-js/src/public/app.ts

```diff
@@ -534,11 +534,16 @@ export namespace app {
    * @returns Promise that will be fulfilled when initialization has completed, or rejected if the initialization fails or times out
    */
   export function initialize(validMessageOrigins?: string[]): Promise<void> {
-    return runWithTimeout(
-      () => initializeHelper(validMessageOrigins),
-      initializationTimeoutInMs,
-      new Error('SDK initialization timed out.'),
-    );
+    if (typeof window !== 'undefined') {
+      return runWithTimeout(
+        () => initializeHelper(validMessageOrigins),
+        initializationTimeoutInMs,
+        new Error('SDK initialization timed out.'),
+      );
+    } else {
+      console.log('window is undefined, continuing without initialization');
+      return Promise.resolve();
```

Comment:
This is going to show my complete naïveté when it comes to SSR usage:
In an SSR scenario, does this code even run? My (likely incorrect) understanding was that in an SSR situation, none of the calls to teamsjs methods actually ever run (because they would all fail since there's no host SDK to answer the calls), they just need to be able to resolve all references/symbols and compile.  Is that true? If that's true, will the log statement here ever be executed?

Reply:
To the best of my knowledge (and testing), you are correct, the log statement does not get run. I decided to include it to cover any potential scenarios that I may have missed/are unaware of in which it may get run, or if teams-js were to extend functionality in the future that would allow it to run on a server. If you feel like it is unnecessary, I am not opposed to removing it.

Reply:
I think I would leave it in but make the message much more dire (and use the logging functions mentioned in another comment). Something like:

"This log statement should NEVER actually be written. This code path exists only to enable compilation in server-side rendering environments. If you EVER see this statement in ANY log file, something has gone horribly wrong and a bug needs to be filed. Peace out."